docs/ai_actions/extend_ai_actions.md

```diff
@@ -0,0 +1,346 @@
+
+You can pass the Action Handler directly to the `ActionServiceInterface::execute()` method, overriding all the other ways of selecting the Action Handler.
+You can also specify the Action Handler by making it part of the passed [Action Configuration](#action-configurations).
+In other cases, the Action Handler is selected automatically. 
+You can influence this choice by creating your own class implementing the [ActionHandlerResolverInterface](../api/php_api/php_api_reference/classes/Ibexa-Contracts-ConnectorAi-Action-ActionHandlerResolverInterface.html) or by listening to the [ResolveActionHandlerEvent](../api/php_api/php_api_reference/classes/Ibexa-Contracts-ConnectorAi-Events-ResolveActionHandlerEvent.html) Event that the default implementation emits.
+
+You can influence the execution of an Action with two events:
+
+- [BeforeExecuteEvent](../api/php_api/php_api_reference/classes/Ibexa-Contracts-ConnectorAi-Action-Event-BeforeExecuteEvent.html), fired before the Action is executed
+- [ExecuteEvent](../api/php_api/php_api_reference/classes/Ibexa-Contracts-ConnectorAi-Action-Event-ExecuteEvent.html), fired after the Action is executed
+
+Below you can find the full example of a Symfony Command, together with a matching service definition. The command finds the images modified in the last 24 hours, and adds the alternative text to them if it's missing.
+
+``` php hl_lines="87 100-125"
+[[= include_file('code_samples/ai_actions/src/Command/AddMissingAltTextCommand.php') =]]
+```
+
+``` yaml
+[[= include_file('code_samples/ai_actions/config/services.yaml', 25, 28) =]]
+```
+
+Executing Actions this way has a major drawback: all the parameters are stored directly in the code and cannot be easily reused or changed. To manage configurations of an AI Action you need to use another concept: Action Configurations.
+
+## Action Configurations
+
+### Manage Action Configurations
+
+Action Configurations allow you to store the parameters for a given Action in the database and reuse them when needed. They can be managed [through the back office]([[= user_doc =]]/ai_actions/work_with_ai_actions.md), through [data migrations](importing_data.md#ai-action-configurations), and through the PHP API.
+
+To manage Action Configurations through the PHP API use the [ActionConfigurationServiceInterface](../api/php_api/php_api_reference/classes/Ibexa-Contracts-ConnectorAi-ActionConfigurationServiceInterface.html) service.
+
+You can manage them using the following methods:
+
+- Create them with `ActionConfigurationServiceInterface::createActionConfiguration()`, passing the [ActionConfigurationCreateStruct](../api/php_api/php_api_reference/classes/Ibexa-Contracts-ConnectorAi-ActionConfiguration-ActionConfigurationCreateStruct.html).
```

Comment:
```suggestion
- Creating them with `ActionConfigurationServiceInterface::createActionConfiguration()`, that passes the [ActionConfigurationCreateStruct](../api/php_api/php_api_reference/classes/Ibexa-Contracts-ConnectorAi-ActionConfiguration-ActionConfigurationCreateStruct.html).
```

Reply:
It's the developer who should pass the `ActionConfigurationCreateStruct` argument to the `ActionConfigurationServiceInterface::createActionConfiguration()` method.
I will modify this one (and the following ones, that are similar) to:
```
Create them with `ActionConfigurationServiceInterface::createActionConfiguration()` by passing the [ActionConfigurationCreateStruct](../api/php_api/php_api_reference/classes/Ibexa-Contracts-ConnectorAi-ActionConfiguration-ActionConfigurationCreateStruct.html).
```